Bottom: 4d91d69621a4849cbb1008e0222430122af2d989
Top:    250d790e7a782b4f26bbd2935aef417ad843968f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 01:14:29 +0100

 Make DrawPPV() always draw for non gui HIDs, rather than doing_assy

The gui HIDs don't accept the assembly drawaing, so testing for !gui->gui
will catch this case, and we can drop the doing_assy override for
PCB->PinOn and PCB->ViaOn


---

diff --git a/src/draw.c b/src/draw.c
index 7345912..1cbc98b 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -566,7 +566,8 @@ pad_callback (const BoxType * b, void *cl)
 }
 
 /* ---------------------------------------------------------------------------
- * draws pins pads and vias
+ * Draws pins pads and vias - Always draws for non-gui HIDs,
+ * otherwise drawing depends on PCB->PinOn and PCB->ViaOn
  */
 static void
 DrawPPV (int group, const BoxType *drawn_area)
@@ -575,7 +576,7 @@ DrawPPV (int group, const BoxType *drawn_area)
   int solder_group = GetLayerGroupNumberByNumber (solder_silk_layer);
   int side;
 
-  if (PCB->PinOn || doing_assy)
+  if (PCB->PinOn || !gui->gui)
     {
       /* draw element pins */
       r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_callback, NULL);
@@ -595,7 +596,7 @@ DrawPPV (int group, const BoxType *drawn_area)
     }
 
   /* draw vias */
-  if (PCB->ViaOn || doing_assy)
+  if (PCB->ViaOn || !gui->gui)
     {
       r_search (PCB->Data->via_tree, drawn_area, NULL, via_callback, NULL);
       r_search (PCB->Data->via_tree, drawn_area, NULL, hole_callback, NULL);
